rfd/0138/README.md

```diff
@@ -0,0 +1,146 @@
+
+
+## Assumptions
+* A TOR switch with DHCP relay capability will be used to route traffic for each of the admin subnets. 
+* The "admin" network pool will only contain a single network per nictag.
+
+
+## Implementation
+One main issue is that most services and configuration scripts require at least one CN nic to have the nictag 'admin'
+
+The convention throughout SDC is to use the format: 
+'<tag>_nic=<mac addr>'
+to denote which nic should be assigned which nictag.
+
+Initially the approach was to simply override 'admin_xxx' with the admin network's properties (i.e. xxx == ip, mac, etc so that admin_nic=<mac address of admin_rack99_nic>).  Unfortunately that won't work because as noted above most of the configuration processing code use the actual property name as a tag name 'foo_nic' for 'foo' nictag.  When we run `nictagadm list` we will see:
+
+```
+NAME           MACADDRESS         LINK           TYPE            
+admin_rack99   d2:9c:54:2c:8c:a4  vioif0         normal          
+admin          d2:9c:54:2c:8c:a4  vioif0         normal   
+```
+
+So any agents looking for the CN's nictags will see a single nic tagged with two different tags.  This may lead other agents and services to believe the CN is on the "admin" subnet(NAPI network), when it is really only on the "admin_rack99" subnet(NAPI network).
+
+This leaves us with two possible approaches:
+a) modify the existing behavior of every script that parses 'admin_nic' as '<nictag>_nic'
+
+OR
+
+b) Add a new property "admin_tag" to denote an alternately tagged admin network (e.g. "admin_tag=admin_rack99".  Then in each script that parses the usbkey or networking.json config we could have the value of "admin_tag" override the string "admin" in the results.  Said another way, if "admin_tag" is set, any configuration file consumers will essentially ignore values of the form "admin_XXX", unless "admin_tag=admin" (which would be the default behavior, and there would be no use in actually setting that...).  So if "admin_tag=admin_rack99" then the configuration file consumers would use values of "admin_rack99" instead of those from "admin".  `nictagadm` would simply skip over anything of the form "admin_XXX".
+
+Currently (b) is the preferred approach because it keeps the changes well partitioned, and backwards compatible.  Also docs that refer to the current property format (<tag name>_nic=<mac addr>) would still be valid. (e.g. https://wiki.smartos.org/display/DOC/extra+configuration+options)
+
+
+There is a general desire to move the CN networking configuration towards being completely contained in the `networking.json` file received from the booter (dhcpd service).  With this in mind, a new object "adminNic" will be added to the networking.json file.  This object will hold all the properties for THE admin NIC for the receiving CN's multi-subnet (or even single subnet) admin network.  
+
+```
```

Comment:
Do we need `"adminNic"` if we have `"admin_tag"`? If we have the latter, then can't we use that to find the right NIC?